samza-log4j/src/main/java/org/apache/samza/logging/log4j/StreamAppender.java

```diff
@@ -300,7 +300,8 @@ protected void setupSystem() {
     String systemFactoryName = log4jSystemConfig.getSystemFactory(systemName)
         .orElseThrow(() -> new SamzaException(
             "Could not figure out \"" + systemName + "\" system factory for log4j StreamAppender to use"));
-    SystemFactory systemFactory = Util.getObj(systemFactoryName, SystemFactory.class);
+    SystemFactory systemFactory =
```

Comment:
Is this the framework classloader? What classloader is the StreamAppender created with? I think this should use the plugin classloader, right?

Reply:
`getClass().getClassLoader()` needs to return the plugin classloader for this to work. So `StreamAppender` needs to be created by the plugin classloader.
I thought I had an understanding around how the logging classloaders worked, but looking again at my notes, I was focusing on log4j, not necessarily slf4j on top of log4j. I will need to do a few more experiments to ensure that `StreamAppender` gets created correctly.
In any case, for this PR, there should be no impact, since it should be the same classloader as before.

Reply:
Just to add a little more context:
`StreamAppender` would get loaded by `Thread.currentThread().getContextClassLoader()`. If the context classloader is unset (can also disable usage of context classloader in log4j), then it will fall back to the classloader of the class which is using the logger.